compiler/vmdef.nim

```diff
@@ -12,17 +12,43 @@
 
 import ast, idents, options, modulegraphs, lineinfos
 
+type TInstrType* = uint64
+
 const
+  regOBits = 8 # Opcode
+  regABits = 16
+  regBBits = 16
+  regCBits = 16
+  regBxBits = 32
```

Comment:
I am not sure if I like the size duplication of everything here. I am not strongly against it, but I think you should have some justification why you increase the size, and some benchmark about the cost (memory usage, and performance). And if this is changed here, we should also take into consideration to add a 4th register argument, as there are some opcodes that scream for a 4th argument.

Reply:
Sure, all valid points.

My goal was to have all sizes clearly defined in one single place so we can easily change the layout to whatever is needed without having to tweak in three places. I have scratched my personal itch with relative jumps by increasing `regBx`, and I have no opinion at all about what to do with the remaining bits. @sealmove indicated that he kept running out of registers, so also increasing `regA`, `regB`, and `regC` seemed like the natural thing to do, but feel free to come up with other or better ideas.

Technically there is no reason to have all sizes set to multiples of 8, although I'm pretty sure the C compiler is much happier when this is the case.

Reply:
I just commented to sealmove that he should check if current `devel` branch has the problem with the vm registers as this situation really has improved thing month. 

What you can also do is to create a giant case object covering all opcodes. This is as far as I know the only safe language pattern that allows opcode dependent fields. But I also think that change would be accepted by Araq, as it is pretty drastic and it causes performance penalties through runtime checks.